packages/yew/src/virtual_dom/vnode.rs

```diff
@@ -45,6 +45,102 @@ impl VNode {
+    ///     <div>
+    /// }
+    /// # }
+    /// ```
+    #[cfg(any(feature = "csr", feature = "hydration", feature = "ssr"))]
+    pub fn from_raw_html(html: &str) -> Self {
+        #[cfg(any(feature = "csr", feature = "hydration"))]
+        fn inner(html: &str) -> VNode {
+            let div = gloo::utils::document().create_element("div").unwrap();
+            div.set_inner_html(html);
+            VNode::VRef(div.into())
+        }
+
+        #[cfg(feature = "ssr")]
+        fn inner(html: &str) -> VNode {
+            use html_parser::{Dom, Node};
+
+            use super::{ApplyAttributeAs, Attributes};
+            use crate::{AttrValue, Classes};
+            fn dom_node_to_vnode(node: Node) -> VNode {
+                match node {
+                    Node::Text(text) => VNode::from(VText::new(text)),
+                    Node::Element(element) => {
+                        let mut tag = VTag::new(element.name);
+                        if !element.attributes.is_empty() {
+                            let attributes = element
+                                .attributes
+                                .into_iter()
+                                .map(|(key, value)| {
+                                    (
+                                        AttrValue::from(key),
+                                        (
+                                            AttrValue::from(value.unwrap_or_default()),
+                                            ApplyAttributeAs::Attribute,
+                                        ),
+                                    )
+                                })
+                                .collect();
+                            tag.set_attributes(Attributes::IndexMap(attributes));
+                        }
+                        if let Some(id) = element.id {
+                            tag.add_attribute("id", id)
+                        }
+                        if !element.classes.is_empty() {
+                            tag.add_attribute("class", Classes::from(element.classes).to_string())
+                        };
+                        tag.add_children(
+                            element
+                                .children
+                                .into_iter()
+                                .map(dom_node_to_vnode)
+                                .collect::<Vec<_>>(),
+                        );
+                        VNode::from(tag)
+                    }
+                    Node::Comment(_) => VNode::default(),
+                }
+            }
+
+            let dom = Dom::parse(html).map(|it| {
+                let vnodes = it
+                    .children
+                    .into_iter()
+                    .map(dom_node_to_vnode)
+                    .collect::<Vec<_>>();
+                VNode::from(VList::with_children(vnodes, None))
+            });
+            // error handling??
+            dom.unwrap()
+        }
+        inner(html)
```

Comment:
This should not only depend on the activated features, but also on the currently active mode that the node is used in, making this a bit more complicated than just overloading a method.

Reply:
It's totally redone now